clients/src/main/java/org/apache/kafka/common/telemetry/internals/MetricKey.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.common.telemetry.internals;
+
+import org.apache.kafka.common.MetricName;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Value object that contains the name and tags for a Metric.
+ */
+public class MetricKey implements MetricKeyable {
+
+    private final String name;
+    private final Map<String, String> tags;
+
+    /**
+     * Create a {@code MetricKey}
+     *
+     * @param name metric name. This should be the telemetry metric name of the metric (the final name
+     *             under which this metric is emitted).
+     */
+    public MetricKey(String name) {
+        this(name, null);
+    }
+
+    /**
+     * Create a {@code MetricKey}
+     *
+     * @param name metric name. This should be the .converted. name of the metric (the final name
+     *             under which this metric is emitted).
+     * @param tags mapping of tag keys to values.
+     */
+    public MetricKey(String name, Map<String, String> tags) {
+        this.name = Objects.requireNonNull(name);
+        this.tags = tags != null ? Collections.unmodifiableMap(tags) : Collections.emptyMap();
+    }
+
+    public MetricKey(MetricName metricName) {
+        this(metricName.name(), metricName.tags());
+    }
+
+    @Override
+    public MetricKey key() {
+        return this;
+    }
+
+    public String getName() {
```

Comment:
nit: we don't use `get` prefix as a convention; should be rename to `name()`

Reply:
I was told the same - but it doesn't seem to be enforced everywhere.   Call for Apache Kafka style guide.

Reply:
There is no checkstyle rule that would fail the build -- its basically up to everybody (especially committer) to keep an eye out for it.

The guidelines (https://kafka.apache.org/coding-guide) are not super clear about it unfortunately (too Scala centric), but I recently double check with a few other committer and there was agreement that not using `get` is the rule)

> Avoid getters and setters - stick to plain vals or vars instead. If (later on) you require a custom setter (or getter) for a var named myVar then add a shadow var myVar_underlying and override the setter (def myVar_=) and the getter (def myVar = myVar_underlying).

Wanna help with a PR to make it more explicit on the guidelines docs?

Reply:
@mjsax - I mentioned in our team meeting today.  A clear style guide seems missing because different commiters seem to do things differently.  I do want to take a crack at it!

Reply:
Done.